homeassistant/components/modbus/sensor.py

```diff
@@ -105,114 +45,34 @@ async def async_setup_platform(
-            entry[CONF_INPUT_TYPE] = entry[CONF_REGISTER_TYPE]
-            del entry[CONF_REGISTER]
-            del entry[CONF_REGISTER_TYPE]
+    if discovery_info is None:  # pragma: no cover
+        return
 
     for entry in discovery_info[CONF_SENSORS]:
-        if entry[CONF_DATA_TYPE] == DATA_TYPE_STRING:
-            structure = str(entry[CONF_COUNT] * 2) + "s"
-        elif entry[CONF_DATA_TYPE] != DATA_TYPE_CUSTOM:
-            try:
-                structure = f">{DEFAULT_STRUCT_FORMAT[entry[CONF_DATA_TYPE]][entry[CONF_COUNT]]}"
-            except KeyError:
-                _LOGGER.error(
-                    "Unable to detect data type for %s sensor, try a custom type",
-                    entry[CONF_NAME],
-                )
-                continue
-        else:
-            structure = entry.get(CONF_STRUCTURE)
+        hub = hass.data[MODBUS_DOMAIN][discovery_info[CONF_NAME]]
+        sensors.append(ModbusRegisterSensor(hub, entry))
 
-        try:
-            size = struct.calcsize(structure)
-        except struct.error as err:
-            _LOGGER.error("Error in sensor %s structure: %s", entry[CONF_NAME], err)
-            continue
-
-        bytecount = entry[CONF_COUNT] * 2
-        if bytecount != size:
-            _LOGGER.error(
-                "Structure request %d bytes, but %d registers have a size of %d bytes",
-                size,
-                entry[CONF_COUNT],
-                bytecount,
-            )
-            continue
-
-        if CONF_REVERSE_ORDER in entry:
-            if entry[CONF_REVERSE_ORDER]:
-                entry[CONF_SWAP] = CONF_SWAP_WORD
-            else:
-                entry[CONF_SWAP] = CONF_SWAP_NONE
-            del entry[CONF_REVERSE_ORDER]
-        if entry.get(CONF_SWAP) != CONF_SWAP_NONE:
-            if entry[CONF_SWAP] == CONF_SWAP_BYTE:
-                regs_needed = 1
-            else:  # CONF_SWAP_WORD_BYTE, CONF_SWAP_WORD
-                regs_needed = 2
-            if (
-                entry[CONF_COUNT] < regs_needed
-                or (entry[CONF_COUNT] % regs_needed) != 0
-            ):
-                _LOGGER.error(
-                    "Error in sensor %s swap(%s) not possible due to count: %d",
-                    entry[CONF_NAME],
-                    entry[CONF_SWAP],
-                    entry[CONF_COUNT],
-                )
-                continue
-        if CONF_HUB in entry:
-            # from old config!
-            hub = hass.data[MODBUS_DOMAIN][entry[CONF_HUB]]
-        else:
-            hub = hass.data[MODBUS_DOMAIN][discovery_info[CONF_NAME]]
-        if CONF_SCAN_INTERVAL not in entry:
-            entry[CONF_SCAN_INTERVAL] = DEFAULT_SCAN_INTERVAL
-        sensors.append(
-            ModbusRegisterSensor(
-                hub,
-                entry,
-                structure,
-            )
-        )
-
-    if not sensors:
-        return
-    async_add_entities(sensors)
+    if len(sensors) > 0:
```

Comment:
It's not necessary to explicitly check length of containers to check if they have any item in Python. This is enough:
```py
if sensors:
```

Reply:
Solved see #51139